src/utils/startNewChallenge/startNewChallenge.function.spec.ts

```diff
@@ -0,0 +1,215 @@
+import { startNewChallenge } from './startNewChallenge.function';
+import { Challenge } from '../../models';
+import { ChallengeStates, StatusStates } from '../../enums';
+import {
+	mockedAchievementStatus,
+	mockedActualAchievements,
+	mockedTasks,
+	mockedTasksStatus
+} from './startNewChallenge.function.mock';
+
+const mockedExpectedChallenge = {
+		id: 2,
+		state: ChallengeStates.InProgress,
+		startDate: new Date().toLocaleDateString(),
+		tasksOrder: mockedTasks,
+		tasksStatus: mockedTasksStatus,
+		duration: 30,
+		archiveItems: [],
+		actualAchievements: mockedActualAchievements,
+		achievementsStatus: mockedAchievementStatus,
+	} as Challenge;
+
+const mockedChallenges: Challenge[] = [
+	{
+		id: 1,
+		state: ChallengeStates.InProgress,
+		startDate: new Date().toLocaleDateString(),
+		tasksOrder: mockedTasks,
+		tasksStatus: mockedTasksStatus,
+		duration: 30,
+		archiveItems: [],
+		actualAchievements: mockedActualAchievements,
+		achievementsStatus: mockedAchievementStatus,
+	} as Challenge,
+];
+
+const mockedFinishedChallenges: Challenge[] = [
+	{
+		id: 1,
+		state: ChallengeStates.Success,
+		startDate: new Date().toLocaleDateString(),
+		tasksOrder: mockedTasks,
+		tasksStatus: mockedTasksStatus,
+		duration: 30,
+		archiveItems: [],
+		actualAchievements: mockedActualAchievements,
+		achievementsStatus: mockedAchievementStatus,
+	} as Challenge,
+];
+
+describe('#startNewChallenge', () => {
+	describe('Arguments', () => {
+		describe('Empty tasks', () => {
+			it('should return empty object, if no tasks list was passed as an argument', () => {
+				const createdChallenge = startNewChallenge([], mockedChallenges, 0, 0);
```

Comment:
If you are using TS, you can check the type of the arguments by TS mechanism and we don't need to test in our test cases, or I missing something?